src/components/Typography/index.tsx

```diff
@@ -1,75 +1,112 @@
 import React, { HTMLAttributes, PropsWithChildren } from 'react'
 
 export type TypographyProps = PropsWithChildren<HTMLAttributes<HTMLDivElement>>
+export type HeadingProps = PropsWithChildren<HTMLAttributes<HTMLHeadingElement>>
 
-export const Heading1 = (props: TypographyProps) => (
-  <h1 {...props}>{props.children}</h1>
+export const Heading1 = (props: HeadingProps) => (
+  <h1 {...props} className={`h1 ${props.className ?? ''}`}>
```

Comment:
you shouldn't need to add h1 as a classname since h1 is already styled with your `h1` css class in _typography.scss